Firebase/Messaging/FIRMessagingRemoteNotificationsProxy.m

```diff
@@ -401,18 +400,13 @@ - (void)application:(UIApplication *)application
 }
 #endif // TARGET_OS_IOS
 
+#if TARGET_OS_IOS || TARGET_OS_TV
 - (void)application:(UIApplication *)application
 didReceiveRemoteNotification:(NSDictionary *)userInfo
 fetchCompletionHandler:(void (^)(UIBackgroundFetchResult))completionHandler {
   [[FIRMessaging messaging] appDidReceiveMessage:userInfo];
 }
 
-- (void)application:(UIApplication *)application
-didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken {
-  // Pass the APNSToken along to FIRMessaging (and auto-detect the token type)
-  [FIRMessaging messaging].APNSToken = deviceToken;
-}
-
 - (void)application:(UIApplication *)application
```

Comment:
The [method](https://developer.apple.com/documentation/appkit/nsapplicationdelegate/1428554-application?language=objc) exists for macOS. `UIApplication` should be replaced by `GULApplication` to keep the functionality.

Reply:
Good catch. Although the method that is swizzled is for analytics to track which is not available for tv/mac. we might need to clean up that in the future.